selfdrive/modeld/runners/onnxmodel.cc

```diff
@@ -87,72 +86,25 @@ void ONNXModel::pread(float *buf, int size) {
   LOGD("host read done");
 }
 
-void ONNXModel::addRecurrent(float *state, int state_size) {
-  rnn_input_buf = state;
-  rnn_state_size = state_size;
+void ONNXModel::addInput(const char *name, float *buffer, int size) {
+  inputs.push_back(ModelInput(name, buffer, size));
 }
 
-void ONNXModel::addDesire(float *state, int state_size) {
-  desire_input_buf = state;
-  desire_state_size = state_size;
-}
-
-void ONNXModel::addNavFeatures(float *state, int state_size) {
-  nav_features_input_buf = state;
-  nav_features_size = state_size;
-}
-
-void ONNXModel::addDrivingStyle(float *state, int state_size) {
-    driving_style_input_buf = state;
-    driving_style_size = state_size;
-}
-
-void ONNXModel::addTrafficConvention(float *state, int state_size) {
-  traffic_convention_input_buf = state;
-  traffic_convention_size = state_size;
-}
-
-void ONNXModel::addCalib(float *state, int state_size) {
-  calib_input_buf = state;
-  calib_size = state_size;
-}
-
-void ONNXModel::addImage(float *image_buf, int buf_size) {
-  image_input_buf = image_buf;
-  image_buf_size = buf_size;
-}
-
-void ONNXModel::addExtra(float *image_buf, int buf_size) {
-  extra_input_buf = image_buf;
-  extra_buf_size = buf_size;
+void ONNXModel::setInputBuffer(const char *name, float *buffer, int size) {
```

Comment:
the setInputBuffer code is mostly the same. can you pull those parts out into a common one?